File: rust/src/n0217_contains_duplicate.test.rs
use crate::n0217_contains_duplicate::*;

#[test]
fn test_has_duplicate() {
    assert!(has_duplicate(&[1, 2, 3, 1]));
    assert!(!has_duplicate(&[1, 2, 3, 4]));
    assert!(has_duplicate(&[1, 1, 1, 3, 3, 4, 3, 2, 4, 2]));
}

#[test]
fn test_has_duplicate_early_return() {
    assert!(has_duplicate_early_return(&[1, 2, 3, 1]));
    assert!(!has_duplicate_early_return(&[1, 2, 3, 4]));
    assert!(has_duplicate_early_return(&[1, 1, 1, 3, 3, 4, 3, 2, 4, 2]));
}

File: typescript/src/n0217_contains_duplicate.test.ts
import { expect } from "jsr:@std/expect";
import {
  hasDuplicate,
  hasDuplicateEarlyReturn,
} from "./n0217_contains_duplicate.ts";

Deno.test("hasDuplicate", () => {
  expect(hasDuplicate([1, 2, 3, 1])).toBe(true);
  expect(hasDuplicate([1, 2, 3, 4])).toBe(false);
  expect(hasDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2])).toBe(true);
});

Deno.test("hasDuplicateEarlyReturn", () => {
  expect(hasDuplicateEarlyReturn([1, 2, 3, 1])).toBe(true);
  expect(hasDuplicateEarlyReturn([1, 2, 3, 4])).toBe(false);
  expect(hasDuplicateEarlyReturn([1, 1, 1, 3, 3, 4, 3, 2, 4, 2])).toBe(true);
});
